### DISCUSSION TITLE
Why Prisma Go is 2x slower than Prisma for Nodejs?

### DISCUSSION THREAD

**Discussion by pstachula-dev** [prisma/prisma-client-go#1229] [Q&A]
I have test with same databse for very simple query:

Go: 43-52ms
```
post, err := client.UserCuid.FindMany().Take((10_000)).Exec(ctx)
```

Nodejs: 26-42ms
```
const posts = await prisma.userCuid.findMany({take:10_000});
```

**Comment by steebchen** [comment]
Good question, most likely because NodeJS is using native bindings whereas the Go client communicates via a http server. It's planned to do the same thing for the Go client but is connected with some effort. Additional optimizations in the parsing etc. might also help but the main reason is most likely due to the differences in the communication between client and the engine.

**Reply by pstachula-dev** [reply]
Ok Thanks for answer, with different ORM like Gorm on my PC I have 6x better results than Prisma. Right now Prisma Go is the best ux for orm in Go comunnity, I will wait for performance updates

**Reply by steebchen** [reply]
Yeah the goal has always been dev UX instead of perf. You can follow this issue for updates on that front https://github.com/steebchen/prisma-client-go/issues/199
